jlm/rvsdg/Transformation.cpp

```diff
@@ -52,13 +54,55 @@ TransformationSequence::Run(
   auto statistics = Statistics::Create(rvsdgModule.SourceFilePath().value());
   statistics->StartMeasuring(rvsdgModule.Rvsdg());
 
-  for (const auto & optimization : Transformations_)
+  size_t numPasses = 0;
+  if (DumpRvsdgDotGraphs_)
+  {
+    DumpDotGraphs(
+        rvsdgModule,
+        statisticsCollector.GetSettings().GetOutputDirectory(),
```

Comment:
Instead of `GetSettings().GetOutputDirectory()`, could we add a method to the `StatisticsCollector` called `GetOrCreateOutputDirectory()`, in which we put the logic for creating the statistics output folder (which is currently in `StatisticsCollector::CreateOutputFile()`?

The current `StatisticsCollector::CreateOutputFile()` always includes the module name and a random string in the filename, so I get the point of making the file path manually instead. It would be nice to only create the output folder if anyone actually attempts to put anything there, though.

Reply:
I fully agree. Done